### PR TITLE
Feature/company future interest countries dataset

### PR DESCRIPTION
**Pull request by dvoong**
### Description of change
Add a new API endpoint for company future interest countries dataset

### Checklist

* [x] Has a new newsfragment been created? Check [changelog/README.md](https://github.com/uktrade/data-hub-leeloo/blob/master/changelog/README.md) for instructions
* [ ] Do any added or updated endpoints appear in the API documentation? See [docs/Maintaining the API documentation.md](https://github.com/uktrade/data-hub-leeloo/blob/develop/docs/Maintaining&#32;the&#32;API&#32;documentation.md) for more details
* [ ] Have any relevant search models been updated?
* [ ] Have any relevant fixtures (`fixtures/test_data.yaml`) been updated?
* [ ] Have any relevant select-/prefetch-related field lists in the views and search apps been updated?
* [ ] Has the admin site been updated (for new models, fields etc.)?
* [ ] Has the README been updated (if needed)?

### CODE REVIEW

**Comment by dvoong** [comment]
> Are we able to extend the existing company data set view rather than add a new one?

I had a chat about this with @currycoder and @mcbhenwood, we decided that two separate datasets, one for a company object and one for its relationships to interest in a country would beeasier to manage than a company dataset where we'd have to nest the lists of countries of interest into the company object.

File: changelog/company/future-countries-of-interest-dataset-api-endpoint.api.md [hunk 1]
```diff
@@ -0,0 +1 @@
+`GET /v4/dataset/companies-future-interest-countries-dataset`: Added future interest countries dataset endpoint to be consumed by data-flow and used in data-workspace.
```
[the remark this answers is not in the document]
Comment (dvoong) [reply changelog/company/future-countries-of-interest-dataset-api-endpoint.api.md]: changes made

File: datahub/dataset/company_future_interest_countries/views.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+from datahub.core.hawk_receiver import (
+    HawkAuthentication,
+    HawkResponseSigningMixin,
+    HawkScopePermission,
+)
+from datahub.dataset.company_future_interest_countries.pagination import \
+    CompanyFutureInterestCountriesDatasetViewCursorPagination
+
+
+class CompanyFutureInterestCountriesDatasetView(HawkResponseSigningMixin, APIView):
```
[the remark this answers is not in the document]
Comment (dvoong) [reply datahub/dataset/company_future_interest_countries/views.py]: changes made

File: datahub/dataset/urls.py [hunk 3]
```diff
@@ -1,6 +1,8 @@
 from django.urls import path
 
 from datahub.dataset.company.views import CompaniesDatasetView
+from datahub.dataset.company_future_interest_countries.views import \
```
Comment (reupen) [inline datahub/dataset/urls.py]: We're following the PEP 8 guideline of avoiding `\` for line continuation and using brackets instead. So this should be something like:

```python
from datahub.dataset.company_future_interest_countries.views import (
    CompanyFutureInterestCountriesDatasetView,
)
```

Comment (dvoong) [reply datahub/dataset/urls.py]: changes made

File: datahub/dataset/company_future_interest_countries/pagination.py [hunk 4]
```diff
@@ -0,0 +1,9 @@
+from rest_framework.pagination import CursorPagination
+
+
+class CompanyFutureInterestCountriesDatasetViewCursorPagination(CursorPagination):
+    """
+    Cursor Pagination for CompanyFutureInterestCountries
+    """
+
+    ordering = ('id', 'future_interest_countries__iso_alpha2_code')
```
[the remark this answers is not in the document]
Comment (reupen) [reply datahub/dataset/company_future_interest_countries/pagination.py]: The new model being added in https://github.com/uktrade/data-hub-leeloo/pull/2181 will replace the current many-to-many field. But could deal with renaming things when the data source is changed, yes.

[the remark this answers is not in the document]
Comment (reupen) [reply datahub/dataset/company_future_interest_countries/pagination.py]: @dvoong The comment above doesn't seem to have been addressed; it's ordering by the company ID now. Did you see the suggested query set above?

**Comment by reupen** [comment]
@dvoong Could you have a look at what has happened with the commits?

image

File: datahub/dataset/company_future_interest_countries/pagination.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (dvoong) [reply datahub/dataset/company_future_interest_countries/pagination.py]: I've made those changes. I also included the id of the future_interest_countries relationship which I think we might want in the dataset.

**Comment by dvoong** [comment]
> @dvoong Could you have a look at what has happened with the commits?
> 
> image

I think there are a lot of commits because I merged the `orgin/master` into this branch to get access to the `datahub.dataset.core.views.BaseDatasetView` which was merged into master after this branch was made. Is there a better way to do this?

**Pull request closed without merging**